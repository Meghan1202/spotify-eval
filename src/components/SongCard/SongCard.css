.Card-Container {
  display: flex;
  flex-direction: column;
  background-color: #dfdfdf;
  width: 15rem;
  height: 15rem;
  border-radius: 1rem;
  padding: 0.625rem;
  margin: 0.3rem;
}

.Image-Container {
  height: 10rem;
  width: 10rem;
  border-radius: 10rem;
  align-self: center;
}

.Song-Name {
  color: black;
  align-self: flex-start;
}
