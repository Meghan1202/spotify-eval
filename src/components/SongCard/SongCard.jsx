import React from 'react';
import PropTypes from 'prop-types';
import './SongCard.css';
import LikeButton from '../LikeButton/LikeButton';

const SongCard = ({ name, albumArt }) => (
  <div className="Card-Container">
    <img className="Image-Container" src={albumArt} alt="poster" />
    <p className="Song-Name">{name}</p>
    <span>Artist</span>
    <LikeButton />
  </div>
);

SongCard.propTypes = {
  name: PropTypes.string.isRequired,
  albumArt: PropTypes.string.isRequired,
};
export default SongCard;
