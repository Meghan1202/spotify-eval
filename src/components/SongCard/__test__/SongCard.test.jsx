import React from 'react';
import { render } from '@testing-library/react';
import SongCard from '../SongCard';

describe(SongCard.name, () => {
  test('Should match the snapshot', () => {
    const { container } = render(<SongCard
      name="songe_Name"
      albumArt="Url"
      artist="Artist_Name"
      currentState
      likes={1}
      id="abc"
      patchLike={jest.fn()}
      updateSongs={jest.fn()}
      theme={1}
    />);
    expect(container).toMatchSnapshot();
  });
});
