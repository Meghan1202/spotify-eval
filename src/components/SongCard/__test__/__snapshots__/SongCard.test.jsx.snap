// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongCard Should match the snapshot 1`] = `
<div>
  <div
    class="Card-Container-1"
  >
    <img
      alt="poster"
      class="Image-Container"
      src="Url"
    />
    <p
      class="Song-Name"
    >
      songe_Name
    </p>
    <span
      class="Artist-Name"
    >
      Artist_Name
    </span>
    <div
      class="Like-Button-Container"
    >
      <input
        alt="Like"
        class="Heart-Button"
        src="./assets/red-heart.svg"
        type="image"
      />
      <div
        class="centered"
      >
        1
      </div>
    </div>
  </div>
</div>
`;
